packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.spec.js

```diff
@@ -1,20 +1,173 @@
+      }).then(done, done)
+    })
+
+    it('should error if the location request response fails', function (done) {
+      wrapper = shallow(
+        <LightCurveViewerContainer.wrappedComponent subject={failSubject} />
+      )
+
+      cdmSpy.returnValues[0].then(() => {
+        expect(wrapper.state().loading).to.equal(asyncStates.error)
+        expect(wrapper.state().error).to.equal('Not Found')
+      }).then(done, done)
+    })
+
+    it('should set the component state with the json data', function (done) {
+      wrapper = shallow(
+        <LightCurveViewerContainer.wrappedComponent subject={subject} />
+      )
+
+      cdmSpy.returnValues[0].then(() => {
+        wrapper.state().dataExtent.x.forEach((xDataPoint, index) => {
+          expect(xDataPoint).to.equal(mockD3XExtent[index])
+        })
+
+        wrapper.state().dataExtent.y.forEach((yDataPoint, index) => {
+          expect(yDataPoint).to.equal(mockD3YExtent[index])
+        })
+
+        expect(wrapper.state().dataPoints).to.have.lengthOf(mockZipData.length)
+
+        expect(wrapper.state().loading).to.equal(asyncStates.success)
+      }).then(done, done)
+    })
+
+    it('calls the onReady prop on successful load', function (done) {
+      const onReadySpy = sinon.spy()
+      wrapper = shallow(
+        <LightCurveViewerContainer.wrappedComponent onReady={onReadySpy} subject={subject} />
+      )
+      
+      cdmSpy.returnValues[0].then(() => {
+        expect(onReadySpy).to.have.been.calledOnceWith({ target: wrapper.instance().viewer.current })
+      })
+      .then(done, done)
+    })
+
+    it('should update component state when there is a new valid subject prop', function (done) {
+      const nextSubjectJSONZip = zip(nextSubjectJSON.x, nextSubjectJSON.y)
+      const nextSubjectD3XExtent = d3.extent(nextSubjectJSON.x)
+      const nextSubjectD3YExtent = d3.extent(nextSubjectJSON.y)
+      const cduSpy = sinon.spy(LightCurveViewerContainer.wrappedComponent.prototype, 'componentDidUpdate')
+      wrapper = shallow(
+        <LightCurveViewerContainer.wrappedComponent subject={subject} />
+      )
+      wrapper.setProps({ subject: nextSubject })
+
+      // componentDidUpdate gets called three times
+      // only the third is the update to the new props we're expecting for this test
+      // Is there a better way to do this? This seems brittle
+      const cduReturnedPromise = cduSpy.returnValues.filter(value => Promise.resolve(value) === value)
+      cduReturnedPromise[0].then(() => {
+        wrapper.state().dataExtent.x.forEach((xDataPoint, index) => {
+          expect(xDataPoint).to.equal(nextSubjectD3XExtent[index])
+        })
+
+        wrapper.state().dataExtent.y.forEach((yDataPoint, index) => {
+          expect(yDataPoint).to.equal(nextSubjectD3YExtent[index])
+        })
+
+        expect(wrapper.state().dataPoints).to.have.lengthOf(nextSubjectJSONZip.length)
+
+        expect(wrapper.state().loading).to.equal(asyncStates.success)
```

Comment:
The loading state should be checked while the new subject is loading too, so immediately after the props change.